src/kibali/PermissionsStubGenerator.cs

```diff
@@ -35,14 +53,145 @@ public string GenerateTable()
+            var higher = scheme.Value.Item2.Split(',').Select(k => k.Trim());
+            if (mergedTableScopes.TryGetValue(scheme.Key, out Dictionary<string, SortedSet<string>> mergedScopes))
+            {
+                mergedScopes["least"].Add(least);
+                mergedScopes["higher"].UnionWith(higher);
+            }
+            else
+            {
+                mergedTableScopes[scheme.Key] = new() { { "least", new() { least } }, { "higher", new SortedSet<string>(higher) } };
+            }
+        }
+    }
+
+    Dictionary<string, (string, string)> GroupAndCleanScopes(Dictionary<string, Dictionary<string, SortedSet<string>>> mergedTableScopes)
+    {
+        var notPresent = new[] { StringConstants.PermissionNotSupported, StringConstants.PermissionNotAvailable };
+        var groupedSchemePermissions = new Dictionary<string, (string, string)>();
+        foreach (var schemeEntry in mergedTableScopes)
+        {
+            var scheme = schemeEntry.Key;
+            var least = schemeEntry.Value["least"];
+            string mergedLeast;
+            if (least.Count == 1)
+            {
+                mergedLeast = least.First();
+            }
+            else
+            {
+                var filtered = least.Except(notPresent);
+                if (filtered.Any())
+                {
+                    if (filtered.Count() > 1)
+                    {
+                        mergedLeast = filtered.OrderBy(m => m).First();
+                        // TODO: Track these cases
+                        ////throw new ArgumentException("Unable to merge least privilege permissions.");
+                    }
+                    else
+                    {
+                        mergedLeast = filtered.First();
+                    }
+                }
+                else
+                {
+                    mergedLeast = StringConstants.PermissionNotSupported;
+                }
+            }
+
+            var higher = schemeEntry.Value["higher"];
+            string mergedHigher;
+            if (higher.Count == 1)
+            {
+                var higherPerm = higher.First();
+                mergedHigher = (higherPerm == mergedLeast && higherPerm != StringConstants.PermissionNotSupported) ? StringConstants.PermissionNotAvailable : higherPerm;
+            }
+            else
+            {
+                least.Remove(mergedLeast); // Remove least privileged permission and add the rest to the least
+                var toSkip = notPresent.Append(mergedLeast);
+                var filtered = higher.Concat(least).Except(toSkip);
+                if (filtered.Any())
+                {
+                    if (filtered.Count() > 1)
+                    {
+                        mergedHigher = string.Join(", ", filtered);
+                    }
+                    else
+                    {
+                        mergedHigher = filtered.First();
+                    }
```

Comment:
```suggestion
                    mergedHigher = string.Join(", ", filtered);
```

Reply:
We can remove the check for 
```csharp
if (filtered.Count() > 1){

```

since join will always return the expected value even when we have 1 element.